profile/preset.go

```diff
@@ -0,0 +1,127 @@
+package profile
+
+import (
+	"crypto"
+
+	"github.com/ProtonMail/go-crypto/openpgp/packet"
+	"github.com/ProtonMail/go-crypto/openpgp/s2k"
+	"github.com/ProtonMail/gopenpgp/v3/constants"
+)
+
+var nameToProfile = map[string]func() *Custom{
+	"default":                              Default,
+	"rfc4880":                              RFC4880,
+	"draft-koch-eddsa-for-openpgp-00":      GnuPG,
+	"draft-ietf-openpgp-crypto-refresh-10": CryptoRefresh,
+}
+
+// PresetProfiles returns the names of the available profiles.
+func PresetProfiles() []string {
+	profiles := make([]string, len(nameToProfile))
+	index := 0
+	for profile := range nameToProfile {
+		profiles[index] = profile
+		index++
+	}
+	return profiles
+}
+
+// Default returns a custom profile that support features
+// that are widely implemented.
+func Default() *Custom {
+	setKeyAlgorithm := func(cfg *packet.Config, securityLevel int8) {
+		cfg.Algorithm = packet.PubKeyAlgoEdDSA
+		switch securityLevel {
+		case constants.HighSecurity:
+			cfg.Curve = packet.Curve25519
```

Comment:
Wow such a difference in security

Reply:
Yes, it is much more secure ;P
Daniel suggested to not include curve448 in the default profile.